File: MIniEp1/miniep1.java
/* miniEP1
 * Nome: Ísis Ardisson Logullo
 * nUSP: 7577410
 * java miniep1.java
 */

import java.util.*;


public class miniep1{
    public static void main(String[] args) {
        double n = 1;
        int i, j, k = 0;
        double bitshift = 0;
        int primos = 0;
        int primosEspeciais = 0;
        

        System.out.println("Digite um numero: ");
        Scanner sc = new Scanner(System.in);
        bitshift = sc.nextDouble();
        sc.close();
        n = n * Math.pow(2,bitshift);
        
        boolean[] crivo = new boolean[(int)n+1];

        for (i = 2; i < n+1; i++){
            crivo[i] = true;
        }
        for (i = 2; i < n+1; i++){
            k = i;
            if (crivo[i] == true){
                primos++;
                if (i % 4 != 3){
                    primosEspeciais++;
                }
                for (j = i*2; j < n+1; k++){
                    crivo[j] = false;
                    j = j + k ;
                }
            }
        }


        System.out.printf("%d %d\n", primos, primosEspeciais);

    }
}

File: MIniEp1/miniep1.c
/* miniEP1
 * NOme: Ísis Ardisson Logullo
 * nUSP: 7577410
 * gcc -Wall -ansi -o MiniEP1 miniep1.c -lm
 */

#include <stdio.h>
#include <math.h>
#include <stdlib.h>

int main (){
    int n = 1;
    int i, j, k = 0;
    int bitshift = 0;
    int primos = 0;
    int primosEspeciais = 0;
    

    printf("Digite um numero: \n");
    scanf("%d", &bitshift);
    n = n * pow(2,bitshift);
    
    int* crivo= malloc((n+1)*sizeof(int));

    for (i = 2; i < n+1; i++){
        crivo[i] = 1;
    }
    for (i = 2; i < n+1; i++){
        k = i;
        if (crivo[i] == 1){
            primos++;
            if (i % 4 != 3){
                primosEspeciais++;
            }
            for (j = i*2; j < n; k++){
                crivo[j] = 0;
                j = j + k;
            }
        }
    }

    printf("%d %d\n", primos, primosEspeciais);

    return 0;
}

